internal/apis/certmanager/types_issuer.go

```diff
@@ -187,6 +187,27 @@ type VaultIssuer struct {
 	// parameter is ignored for plain HTTP protocol connection. If not set the
 	// system root certificates are used to validate the TLS connection.
 	CABundle []byte
+
+	// Like CABundle but loads the CA bundle from a Kubernetes Secret.
+	// CABundle will take precedence if also set.
+	// +optional
+	CABundleSecretRef VaultCaBundleSecretRef
+}
+
+// VaultCaBundleSecretRef configures a Vault CA bundle via a Kubernetes Secret.
+type VaultCaBundleSecretRef struct {
+	// The name of the Secret resource being referred to.
+	Name string `json:"name"`
+
+	// The namespace of the Secret resource being referred to.
+	// If omitted, it will use the namespace of the referent. If the referent
+	// is a cluster-scoped resource (e.g. a ClusterIssuer), the namespace of
+	// the configured 'cluster resource namespace' will be used. It is set as a
+	// flag on the controller component (and defaults to the namespace that cert-manager runs in).
+	Namespace string `json:"namespace,omitempty""`
```

Comment:
We should never allow referencing source data from arbitrary Namespaces. This would allow users to read resources in Namespaces they don't necessarily have access to with RBAC.

Issuer resources always source data from the namespace they reside. ClusterIssuers source data always from the cluster resource namespace.

We should change `VaultCaBundleSecretRef` to be [`SecretKeySelector`](https://github.com/cert-manager/cert-manager/blob/6947696b12428a1f9e5823c2e69de84b8dd08c1c/pkg/apis/meta/v1/types.go#L65).

Reply:
SecretKeySelector allows for the secret key to be empty. I can see that SecretKeySelector is also being used in other cases where the key actually must not be empty. Why not use a dedicated key selector type that makes the secret key mandatory?

Reply:
Whilst having two types for a required and non-required key field may be beneficial, it's likely not worth adding another type now as we already have resources using them in both ways.

We can't change existing resources as this would break typing, which we can't do now we are `v1`.

Continuing to check during validation is probably fine.  We may want to default to `ca.crt`.